contracts/src/PermissionedStakeTable.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { BN254 } from "bn254/BN254.sol";
+import { EdOnBN254 } from "./libraries/EdOnBn254.sol";
+
+/**
+ * @title SimpleStakeTable
+ * @dev An stake table mapping with owner-only access control.
+ */
+contract PermissionedStakeTable is Ownable {
+    event StakersUpdated(NodeInfo[] removed, NodeInfo[] added);
+
+    error StakerAlreadyExists(BN254.G2Point);
+    error StakerNotFound(BN254.G2Point);
+
+    struct NodeInfo {
+        /// The consensus signing key
+        BN254.G2Point blsVK;
+        /// The consensus signing key. Only used for storage in this contract.
+        EdOnBN254.EdOnBN254Point schnorrVK;
+        /// Is the Node DA Node? Only used for storage in this contract.
+        bool isDA;
+    }
+
+    // State mapping from staker IDs to their staking status
+    mapping(bytes32 nodeID => bool isStaker) private stakers;
+
+    constructor(NodeInfo[] memory initialStakers) Ownable(msg.sender) {
+        insert(initialStakers);
+    }
+
+    // public methods
+
+    function update(NodeInfo[] memory stakersToRemove, NodeInfo[] memory newStakers)
```

Comment:
@alysiahuggins only the `update` method is public.

Reply:
ah missed that, thanks